### PR TITLE
fix: tentative fix for test gc_after_sync1

### PR DESCRIPTION
**Pull request by Trisfald**
Quoting the test description:

> Spin up one validating node and one nonvalidating node
> stop the nonvalidating node in the second epoch and
> restart it in the fourth epoch to trigger state sync
> Check that after 10 epochs the node has properly garbage
> collected blocks.

After 10 epochs the test checks that the 6 epochs are garbage collected. The problem is: one particular block (89 - the last block of epoch before state sync) is explicitly requested by state sync so it will be returned by the RPC call.

This PR changes the test to expect the presence of block 89.
If instead the block shouldn't be there, it's a bug in state sync / GC.

### CODE REVIEW

**Comment by Trisfald** [comment]
@VanBarbascu I added you as reviewer, if you can have a look if the updated test assertions are good.

TL;DR - for state sync to work starting at epoch 4 (height 90) it has to fetch the last block of previous epoch (height 89) and that's why block 89 is not GC'd in this scenario

**Comment by Longarithm** [comment]
I suspect this is because block is staying in the cache - and cache is not cleaned up properly, as stated in https://github.com/near/nearcore/blob/9932e95c5c49262189bc4737366ce46bb2b953c8/chain/chain/src/store/mod.rs#L1245-L1259.
Should we rather clean up block-related caches for the sake of data consistency?

**Comment by shreyan-gupta** [comment]
Had a quick chat with Razvan and ideally the block should have been GC'd as we only care about the last block of the previous epoch for state sync purposes and no other block. So the error is a genuine error.

**Comment by Trisfald** [comment]
Thanks, then I'll close this PR since the error is genuine and we should fix it.
